checker/src/abstract_value.rs

```diff
@@ -762,7 +763,7 @@ impl Path {
     pub fn refine_parameters(&self, arguments: &[(Path, AbstractValue)]) -> Path {
         match self {
             Path::LocalVariable { ordinal } if 0 < *ordinal && *ordinal <= arguments.len() => {
-                arguments[ordinal - 1].0.clone()
+                arguments[*ordinal - 1].0.clone()
```

Comment:
Not so much an undocumented assumption as a rather subtle call to a virtual function when a simple subtraction is expected.